core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -167,7 +167,9 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
               pubSub.eventTopic(entityType, slice) ! Topic.Subscribe(ref.toTyped[EventEnvelope[Event]])
             }
           }
-      dbSource.merge(pubSubSource).via(deduplicate(settings.querySettings.deduplicateCapacity))
+      dbSource
+        .mergePrioritized(pubSubSource, leftPriority = 1, rightPriority = 10)
+        .via(deduplicate(settings.querySettings.deduplicateCapacity))
```

Comment:
Did look a bit at Merge trying to figure out if there is something wrt fairness, but nothing obviously unfair in there.

Reply:
I tested some more locally and I indeed see that this helps, but I think the change that had biggest impact in the real testing was that I increased the number of projection instances from 4 to 16 (still running 4 ActorSystems). In that way there are less heavy bursts of events from the db queries.

I also played with the idea of adding `dbSource.throttle`. While that helps it would be difficult to find an optimal rate. Too low and there is a risk that the db queries fall behind the actual produced rate. Too high and it will not matter for fairness. Too low it will also suffer for the raw catch up throughput. No, wrong path.